pydantic/utils.py

```diff
@@ -83,6 +83,15 @@ def in_ipython() -> bool:
         return True
 
 
+def deep_update(mapping: Dict[str, Any], updating_mapping: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
I think I've spotted another problem with `deep_update`, it **updates** `mapping`, rather than building a new dict, this is different from `{**self._build_environ(), **init_kwargs}` and I think not the expected behaviour.

Reply:
I imagined it as a modification of `dict.update` method, which is mutating, but I modified it to return a copy, to be more in accordance with the code that is being replaced.